src/main/java/htsjdk/samtools/cram/structure/Container.java

```diff
@@ -335,6 +336,28 @@ public List<SAMRecord> getSAMRecords(
     public CompressionHeader getCompressionHeader() { return compressionHeader; }
     public AlignmentContext getAlignmentContext() { return containerHeader.getAlignmentContext(); }
     public long getContainerByteOffset() { return containerByteOffset; }
+
+    /**
+     * Update the stream-relative values (global record counter and stream byte offset) for this
+     * container. For use when re-serializing a container that has been read from an existing stream
+     * into a new stream. This method mutates the container and it's slices - the container is no
+     * longer valid in the context of it's original stream.
+     *
+     * @param containerRecordCounter the new global record counter for this container
+     * @param streamByteOffset the new stream byte offset counter for this container
+     * @return the updated global record counter
+     */
+    public long relocateContainer(final long containerRecordCounter, final long streamByteOffset) {
```

Comment:
Is it necessary to mutate the container? It was previously effectively immutable wasn't it?  It seems like it would be nicer to copy it since I assume the container / slice objects are fairly lightweight and you could just pass through the reference to the heavy data?

Not a blocker if you think this is necessary for efficiency reasons / copying a container being very annoying.